plutus-core/plutus-ir/test/PlutusIR/Transform/LetFloatOut/Tests.hs

```diff
@@ -64,3 +68,13 @@ test_letFloatOut = runTestNestedIn ["plutus-ir/test/PlutusIR/Transform"] $
         _ <- runQuoteT . flip inferType (() <$ pirFloated) =<< TC.getDefTypeCheckConfig ()
         -- letmerge is not necessary for floating, but is a nice visual transformation
         pure $ LetMerge.letMerge pirFloated
+
+-- | Check that a term typechecks after a
+-- `PlutusIR.Transform.LetFloatOut.floatTerm` pass.
+typecheck_floatTerm_prop :: Property
+typecheck_floatTerm_prop =
+  pure_typecheck_prop $ LetFloatOut.floatTerm def
```

Comment:
here you're not trying both builtin semantics variants! let's be consistent

Reply:
Since we have some evidence of different results with different variants, I'm gonna try both for all tests.

Reply:
Maybe you're catching my drift by now, which is: just generate all the arguments, configuration included!